ios/browser/ui/webui/brave_wallet/blockchain_images_source.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_IOS_BROWSER_UI_WEBUI_BRAVE_WALLET_BLOCKCHAIN_IMAGES_SOURCE_H_
+#define BRAVE_IOS_BROWSER_UI_WEBUI_BRAVE_WALLET_BLOCKCHAIN_IMAGES_SOURCE_H_
+
+#include <string>
+
+#include "base/files/file_path.h"
+#include "base/memory/weak_ptr.h"
+#include "brave/components/brave_wallet/browser/blockchain_images_source_base.h"
+#include "ios/web/public/webui/url_data_source_ios.h"
+
+namespace brave_wallet {
+
+// This serves background image data.
+class BlockchainImagesSource : public web::URLDataSourceIOS,
+                               public BlockchainImagesSourceBase {
+ public:
+  explicit BlockchainImagesSource(const base::FilePath& base_path);
```

Comment:
You can do `using BlockchainImagesSourceBase::BlockchainImagesSourceBase` to avoid redefining a constructor for this subclass I think?